core-ui/src/components/Extensibility/contexts/DataSources.tsx

```diff
@@ -0,0 +1,209 @@
+import pluralize from 'pluralize';
+import { createContext, useEffect, useRef, FC } from 'react';
+import { useFetch } from 'shared/hooks/BackendAPI/useFetch';
+import { useObjectState } from 'shared/useObjectState';
+import * as jp from 'jsonpath';
+import { jsonataWrapper } from '../helpers/jsonataWrapper';
+
+export interface Resource {
+  metadata: {
+    name: string;
+    namespace: string;
+    labels: { [key: string]: string };
+    annotations: { [key: string]: string };
```

Comment:
```suggestion
    labels: Record<string, string>;
    annotations: Record<string, string>;
```
Maybe that's a little bit nicer?

Reply:
And probably in other places.